### PR TITLE
prebuild.sh to python

### PR DESCRIPTION
**Pull request by noway421**
Instead of relying on a bash script, rewrote it to python. 

This is a prep for release.

### CODE REVIEW

**Comment by feikname** [comment]
LGTM

File: setup.py [hunk 1]
```diff
@@ -38,12 +41,51 @@
...
+        os.remove("pyenet/enet.so")
+    if os.path.isdir("pyenet/enet"):
+        shutil.rmtree("pyenet/enet")
+
+    shutil.copyfile("pyenet/enet.pyx", "pyenet/enet-pyspades.pyx")
+    subprocess.Popen(['patch', '-p1', 'pyenet/enet-pyspades.pyx', 'pyspades-pyenet.patch']).communicate()
+
+    if not os.path.isfile(enet_file):
+        print("Downloading enet")
+        urllib_request.urlretrieve(enet_url, enet_file)
```
Comment (NotAFile) [inline setup.py]: also, we should note that this means that there is no longer a progress bar for the download. I don't think it's a huge problem.

Comment (NotAFile) [inline setup.py]: Turns out it's pretty simple to add a progress bar: https://stackoverflow.com/questions/13881092/download-progressbar-for-python-3

File: .gitignore [hunk 2]
```diff
@@ -53,3 +54,6 @@ enet/pyenet/enet-pyspades.pyx
 enet/pyenet/enet/
 enet/pyenet/build/
 enet/enet-*.tar.gz
+
+#misc
+.DS_Store
```
Comment (feikname) [inline .gitignore]: This line should end with a newline character.

File: setup.py (hunk 1 above)
Comment (noway421) [inline setup.py]: it complicates the script, I would prefer to keep it as lean as possible. "Downloading enet" is quite descriptive and doesn't take too long for user to keep wondering what's going on. The archive is relatively small

**Comment by feikname** [comment]
@NotAFile I converted your requested change to an [issue](https://github.com/piqueserver/piqueserver/issues/36)

**Comment by feikname** [comment]
Just a note that wasn't included here, this fixes #31.

**Pull request merged**